website_cms/tests/test_media.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+import openerp
+
+
+class TestMedia(openerp.tests.HttpCase):
+    """ Test cms.media behavior."""
+
+    at_install = False
+    post_install = True
+
+    def setUp(self):
+        super(TestMedia, self).setUp()
+
+        self.cms_media = self.env['cms.media']
+        # Media 1
+        self.blob1_b64 = 'blob1'.encode('base64')
+        self.f1 = self.cms_media.create(
+            {'name': 'f1', 'datas': self.blob1_b64})
+        # Media 1
+        self.blob2_b64 = 'blob2'.encode('base64')
+        self.f2 = self.cms_media.create(
+            {'name': 'f2',
+             'datas': self.blob1_b64,
+             'datas_fname': 'f2.txt'})
+
+        self.all_files = [self.f1, self.f2]
+
+    def tearDown(self):
+        super(TestMedia, self).tearDown()
+        self.registry('cms.media').unlink(
+            self.cr, 1, [x.id for x in self.all_files])
+
+    def test_url(self):
+        # no file name, default to /download
+        expected = (
+            u'/web/content/cms.media/'
+            u'{}/datas/download').format(self.f1.id)
+        self.assertEqual(self.f1.website_url, expected)
+
+        self.f1.datas_fname = 'foo.pdf'
+        self.f1.invalidate_cache()
+        expected = (
+            u'/web/content/cms.media/'
+            u'{}/datas/foo.pdf').format(self.f1.id)
+        self.assertEqual(self.f1.website_url, expected)
+
+    def test_download_file(self):
+        self.authenticate('admin', 'admin')
+        self.assertTrue(self.f2.exists())
+        # resp = self.url_open(self.f2.website_url)
+
+        # XXX: something really weird is going on here
+        # the former assertion on `exists` passes
+        # but when ir_http.binary_content is called
+        # to dispatch the file content
+        # `exists` returns false and you get a 404?????
```

Comment:
I've tried that as you can see some lines below.... but I can try it again... then maybe I can still poke you :)